internal/controller/serverbootconfiguration_pxe_controller.go

```diff
@@ -177,20 +192,100 @@ func (r *ServerBootConfigurationPXEReconciler) getSystemIPFromBootConfig(ctx con
+func (r *ServerBootConfigurationPXEReconciler) getImageDetailsFromConfig(ctx context.Context, config *metalv1alpha1.ServerBootConfiguration) (string, string, string, error) {
 	imageDetails := strings.Split(config.Spec.Image, ":")
 	if len(imageDetails) != 2 {
 		return "", "", "", fmt.Errorf("invalid image format")
 	}
-	kernelURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerName=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], ironcoreimage.KernelLayerMediaType)
-	initrdURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerName=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], ironcoreimage.InitRAMFSLayerMediaType)
-	// TODO: move this const to ironcore-image
-	const squashFSMediaTypeLayer = "application/vnd.ironcore.image.squashfs.v1alpha1.squashfs"
-	squashFSURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerName=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], squashFSMediaTypeLayer)
+
+	kernelDigest, initrdDigest, squashFSDigest, err := r.getLayerDigestsFromNestedManifest(ctx, imageDetails[0], imageDetails[1])
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to fetch layer digests: %w", err)
+	}
+
+	kernelURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerDigest=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], kernelDigest)
+	initrdURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerDigest=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], initrdDigest)
+	squashFSURL := fmt.Sprintf("%s/image?imageName=%s&version=%s&layerDigest=%s", r.IPXEServiceURL, imageDetails[0], imageDetails[1], squashFSDigest)
 
 	return kernelURL, initrdURL, squashFSURL, nil
 }
 
+func (r *ServerBootConfigurationPXEReconciler) getLayerDigestsFromNestedManifest(ctx context.Context, imageName, version string) (string, string, string, error) {
+	repo, err := orasregistry.NewRepository(imageName)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to create remote repository client: %w", err)
+	}
+
+	store := orasmemory.New()
+	desc, err := oras.Copy(ctx, repo, version, store, version, oras.DefaultCopyOptions)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to fetch index manifest: %w", err)
+	}
+
+	indexManifest, err := store.Fetch(ctx, desc)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to fetch index manifest content: %w", err)
+	}
+
+	defer indexManifest.Close()
+	var nestedManifestDigest string
+	var ociIndexManifest OCIIndexManifest
+	if err := json.NewDecoder(indexManifest).Decode(&ociIndexManifest); err != nil {
+		return "", "", "", fmt.Errorf("failed to decode index manifest: %w", err)
+	}
+
+	for _, layer := range ociIndexManifest.Manifests {
+		if annotations := layer.Annotations; annotations != nil {
+			if strings.HasPrefix(annotations["cname"], "metal_pxe") && layer.Platform.Architecture == "amd64" {
```

Comment:
Do we drop ARM support here?

Reply:
ARM support would need API enhancement in the `Server` CR, to be discussed next week. 
I will consider a switch/flag for a short term to support ARM as well, thanks for pointing it out.

Reply:
https://github.com/ironcore-dev/metal-operator/issues/226 for the follow-up.